Userland/Libraries/LibGfx/ImageFormats/CCITTDecoder.cpp

```diff
@@ -420,20 +420,42 @@ struct CCITTStatus {
     bool has_reached_eol { false };
 };
 
-ErrorOr<void> ensure_invalid_result_is_actually_eol(BigEndianInputBitStream& input_bit_stream, InvalidResult partially_read_eol)
+ErrorOr<void> ensure_invalid_result_is_actually_eol(BigEndianInputBitStream& input_bit_stream, InvalidResult partially_read_eol, Group4Options const& options)
 {
     if (partially_read_eol != 0)
         return Error::from_string_literal("CCITTDecoder: Unable to find the correct mode");
 
     auto const remaining_eol = TRY(input_bit_stream.read_bits(5));
-    if (remaining_eol != 1) {
+    if (options.has_end_of_block == Group4Options::HasEndOfBlock::Yes
+        && remaining_eol == 0) {
+        // Some PDF like 00000337.pdf ends with an EOFB [1] that is byte aligned. This is
+        // what we are trying to detect/read here. As we already read 12 bytes from
+        // partially_read_eol and remaining_eol, we need to realign ourselves first.
+        // [1] 2.4.1.1 End-of-facsimile block
+
+        static constexpr u32 EOFB = 0x001001;
+
+        u8 fill_bits_length = (12 + input_bit_stream.bits_until_next_byte_boundary()) % 8;
+        u8 to_read = fill_bits_length + 12;
+        auto potential_eofb = TRY(input_bit_stream.read_bits(to_read));
+
+        // We already checked that the 12 first bits were zeroes, so here we check that the
+        // last to_read bits end like an EOFB.
```

Comment:
s/like/look like/, probably?

Reply:
Ah no, I misread. "the bits end like". Maybe "that the last to_read bits end with EOFB"? But this is also a nit now. Feel free to do none of the three nits (or as many you'd like) :)